apps/dashboard/src/components/workflow-editor/steps/in-app/in-app-tabs.tsx

```diff
@@ -0,0 +1,143 @@
+import { RiEdit2Line, RiPencilRuler2Line } from 'react-icons/ri';
+import { Cross2Icon } from '@radix-ui/react-icons';
+import { useNavigate, useParams } from 'react-router-dom';
+import { useForm } from 'react-hook-form';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { type WorkflowResponseDto, type StepDataDto, type StepUpdateDto } from '@novu/shared';
+
+import { Form } from '@/components/primitives/form/form';
+import { Notification5Fill } from '@/components/icons';
+import { Button } from '@/components/primitives/button';
+import { Separator } from '@/components/primitives/separator';
+import { Tabs, TabsContent, TabsList, TabsTrigger } from '@/components/primitives/tabs';
+import { InAppEditorPreview } from '@/components/workflow-editor/steps/in-app/in-app-editor-preview';
+import { useUpdateWorkflow } from '@/hooks/use-update-workflow';
+import { buildDynamicZodSchema, buildDefaultValues } from '@/utils/schema';
+import { InAppEditor } from '@/components/workflow-editor/steps/in-app/in-app-editor';
+import { showToast } from '@/components/primitives/sonner-helpers';
+import { ToastIcon } from '@/components/primitives/sonner';
+
+const tabsContentClassName = 'h-full w-full px-3 py-3.5';
+
+export const InAppTabs = ({ workflow, step }: { workflow: WorkflowResponseDto; step: StepDataDto }) => {
```

Comment:
This is the main component that is rendered in the In-App drawer, it consists of the editor and preview tabs, react-hook-form and submission logic.